File: src/test/java/br/com/jiratorio/factory/domain/JiraChangelogItemFactory.java
package br.com.jiratorio.factory.domain;

import br.com.jiratorio.domain.changelog.JiraChangelogItem;
import br.com.leonardoferreira.jbacon.JBacon;
import com.github.javafaker.Faker;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.Date;
import java.util.concurrent.TimeUnit;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class JiraChangelogItemFactory extends JBacon<JiraChangelogItem> {

    @Autowired
    private Faker faker;

    @Override
    protected JiraChangelogItem getDefault() {
        JiraChangelogItem jiraChangelogItem = new JiraChangelogItem();

        jiraChangelogItem.setField("duedate");
        jiraChangelogItem.setTo(toLocalDateTime(faker.date().future(3, TimeUnit.DAYS))
                .format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
        jiraChangelogItem.setCreated(toLocalDateTime(faker.date().past(3, TimeUnit.DAYS)));

        return jiraChangelogItem;
    }

    @Override
    protected JiraChangelogItem getEmpty() {
        return new JiraChangelogItem();
    }

    @Override
    protected void persist(final JiraChangelogItem jiraChangelogItem) {
        throw new UnsupportedOperationException();
    }

    private LocalDateTime toLocalDateTime(final Date date) {
        return date.toInstant()
                .atZone(ZoneId.systemDefault())
                .toLocalDateTime();
    }

}

File: src/test/java/br/com/jiratorio/factory/domain/JiraChangelogItemFactory.kt
package br.com.jiratorio.factory.domain

import br.com.jiratorio.domain.changelog.JiraChangelogItem
import br.com.jiratorio.extension.format
import br.com.jiratorio.extension.toLocalDateTime
import br.com.leonardoferreira.jbacon.JBacon
import com.github.javafaker.Faker
import org.springframework.stereotype.Component
import java.util.concurrent.TimeUnit

@Component
class JiraChangelogItemFactory(
        private val faker: Faker
) : JBacon<JiraChangelogItem>() {

    override fun getDefault() =
            JiraChangelogItem().apply {
                field = "duedate"
                to = faker.date().future(3, TimeUnit.DAYS).format("yyyy-MM-dd")
                created = faker.date().past(3, TimeUnit.DAYS).toLocalDateTime()
            }

    override fun getEmpty() =
            JiraChangelogItem()

    override fun persist(jiraChangelogItem: JiraChangelogItem) =
            throw UnsupportedOperationException()

}
